tests/test_command.py

```diff
@@ -23,6 +29,15 @@
 
 PYTHON = sys.executable
 
+skip_on_windows_console = pytest.mark.skipif(
+    platform.system() == "Windows" and "DETACHED_FROM_CONSOLE" not in os.environ,
+    reason="Do not run this test attached to a Windows console.",
+)
```

Comment:
The wording here is possibly misleading. `CREATE_NO_WINDOW` and `CREATE_NEW_CONSOLE` run the process attached to a new console session. The creation flag `DETACHED_PROCESS`, on the other hand, tells the system to spawn a process detached from any console session. A truly detached process is definitely not wanted here since the test needs to generate and receive a console control event.

FYI, in Windows 8.1+, a console session is a set of processes that are connected to an instance of conhost.exe by way of the ConDrv kernel device. The session window is optional and may be excluded because of `CREATE_NO_WINDOW` or, in Windows 10, because the console host is running in headless conpty mode (pseudoconsole). For example, each tab in Windows Terminal uses a headless conpty session, with an open-source build of "conhost.exe" that's named "openconsole.exe".

Reply:
Thank you for pointing that out. d6b0e59 changes the naming, hopefully for the better :wink: